src/lookup.rs

```diff
@@ -436,6 +474,101 @@ pub(super) async fn fetch_channel_updates<L: Deref>(delta_set: &mut DeltaSet, cl
 	log_info!(logger, "Processed intermediate rows ({}) (delta size: {}): {:?}", intermediate_update_count, delta_set.len(), start.elapsed());
 }
 
+pub(super) async fn fetch_node_updates<L: Deref>(delta_set: &mut NodeDeltaSet, client: &Client, last_sync_timestamp: u32, logger: L) where L::Target: Logger {
```

Comment:
Is there a case where this would be called more than once with the same `NodeDeltaSet`? If not, could we return that instead using an in-out parameter? It's harder to reason about the logic here if it doesn't always start as empty.